packages/stats/gbstats/bayesian/tests.py

```diff
@@ -234,3 +239,356 @@ def get_risk(mu, sigma) -> List[float]:
         risk_ctrl = float((1.0 - prob_ctrl_is_better) * mn_pos)
         risk_trt = -float(prob_ctrl_is_better * mn_neg)
         return [risk_ctrl, risk_trt]
+
+
+@dataclass
+class MidExperimentPowerResult:
+    additional_n: Optional[float]
+    additional_days: Optional[float]
+    update_message: str
+    error: Optional[str] = None
+    power: float = 0.8
+    v_prime: float = 0
+
+
+@dataclass
+class ScalingFactorResult:
+    scaling_factor: Optional[float]
+    converged: bool = False
+    error: Optional[str] = None
+
+
+@dataclass
+class MidExperimentPowerConfig:
+    power: float = 0.8
+    m_prime: float = 1
+    v_prime: float = 1
+    sequential: bool = False
+    sequential_tuning_parameter: float = 5000
+
+
+class CalculateRequiredSampleSize:
+    def __init__(
+        self,
+        stat_a: TestStatistic,
+        stat_b: TestStatistic,
+        config: EffectBayesianConfig,
```

Comment:
You are right, the Bayesian config has superfluous attributes.  We still need attributes like `power`, `m_prime`, `v_prime`, etc., so I kept `MidExperimentPowerConfig`.  

I moved alpha to base config, and `MidExperimentPowerConfig` is now a subclass of `BaseConfig`.

Reply:
> and MidExperimentPowerConfig is now a subclass of BaseConfig.

You could also keep the configs separate and use BaseConfig and MidExperimentPowerConfig. I think that's better since they serve fundamentally different purposes. But it's nbd.

Reply:
good call, and done